marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallBatcherTest.java

```diff
@@ -0,0 +1,336 @@
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.query.DeleteQueryDefinition;
+import com.marklogic.client.query.QueryManager;
+import com.marklogic.client.test.Common;
+
+public class CallBatcherTest {
+	  private final static String ENDPOINT_DIRECTORY = "/javaApi/test/callManager/";
+	  private static ObjectMapper objectMapper = new ObjectMapper();
+	  private DatabaseClient db      = Common.connect();
+	  private CallManager    callMgr = CallManager.on(db);
+	  private JacksonHandle serviceHandle;
+	  private final static Map<String, Format> NODE_FORMATS = new HashMap<>();
+	  private static Map<String, JsonNode> endpointdefs = new HashMap<>();
+	    {
+	        ObjectNode servicedef = objectMapper.createObjectNode();
+	        servicedef.put("endpointDirectory", ENDPOINT_DIRECTORY);
+	        serviceHandle = new JacksonHandle(servicedef);
+	    }
+
+	  
+	  private CallManager.CallableEndpoint installEndpoint(String functionName) {
+		  if(functionName == null || functionName.length() == 0)
+			  throw new MarkLogicInternalException("Invalid input was sent.");
+		  JsonNode endpointdef = endpointdefs.get(functionName);
+	      return callMgr.endpoint(serviceHandle, new JacksonHandle(endpointdef), "sjs");
+	  }
+	  
+	  @BeforeClass
+	  public static void setup() {
+		NODE_FORMATS.put("array", Format.JSON);
+		NODE_FORMATS.put("binaryDocument", Format.BINARY);
+		NODE_FORMATS.put("jsonDocument", Format.JSON);
+		NODE_FORMATS.put("object", Format.JSON);
+		NODE_FORMATS.put("textDocument", Format.TEXT);
+		NODE_FORMATS.put("xmlDocument", Format.XML);
+
+		DatabaseClient adminClient = Common.newServerAdminClient("java-unittest-modules");
+		JSONDocumentManager docMgr = adminClient.newJSONDocumentManager();
+
+		DocumentMetadataHandle docMeta = new DocumentMetadataHandle();
+
+		docMeta.getPermissions().add("rest-reader", DocumentMetadataHandle.Capability.EXECUTE);
+		setupNoParamReturnEndpoint(docMgr, docMeta, "noParamReturn", "double", "1.2");
+		setupEndpointSingleRequired(docMgr, docMeta, "oneParam", "double");
+		setupEndpointMultipleRequired(docMgr, docMeta, "float", "manyParam");
+		
+		adminClient.release();
+	  }
+	  
+	  @Test
+	  public void noneForArgsTest() {
+		  CallManager.CallableEndpoint callableEndpoint = installEndpoint("noParamReturn");
+		  CallManager.NoneCaller noneCaller = callableEndpoint.returningNone();
+		  
+		  CallBatcher<CallManager.CallArgs,CallManager.CallEvent> batcher = noneCaller.batcher().forArgs();
+		  assertNotNull(batcher);
+		  ArrayList<String> assignedParams = new ArrayList<String>();
+		  batcher.onCallSuccess(event-> { 
+			  assignedParams.add(event.getArgs().getAssignedParamNames()[0]);
+		  });
+		  
+		  batcher.getDataMovementManager().startJob(batcher);
+		  
+	      batcher.add(noneCaller.args().param("param1", 1.2));
+
+	      batcher.flushAndWait();
+	      assertTrue(assignedParams.size()!=0);
```

Comment:
Could the assertion be more precise?  Would the test result be valid if the array had more than 1 item?

Also, could the assertion be more informative if it fails?  As in:

`    assertEqual("invalid number of parameters", assignedParams.size(), 1); 
`
The same comment applies to the other tests.